packages/roosterjs-editor-plugins/lib/Paste/officeOnlineConverter/convertPastedContentFromWordOnline.ts

```diff
@@ -0,0 +1,233 @@
+                // insert the current list into body and then reinitialize the convertedListElement
+                const itemLevel = parseInt(item.getAttribute('data-aria-level'));
+                if (convertedListElement.tagName.toUpperCase() != listType && itemLevel == 1) {
+                    insertConvertedListToDoc(convertedListElement, doc.body, itemBlock);
+                    convertedListElement = document.createElement(listType);
+                }
+                insertListItem(convertedListElement, item, listType);
+            });
+        });
+
+        insertConvertedListToDoc(convertedListElement, doc.body, itemBlock);
+
+        const parentContainer = itemBlock.startElement.parentElement;
+        if (parentContainer) {
+            itemBlock.listItemContainers.forEach((listItemContainer) => {
+                parentContainer.removeChild(listItemContainer);
+            });
+        }
+    });
+}
+
+/**
+ * Take all the list items in the document, and group the consecutive list times in a list block;
+ * @param doc pasted document that contains all the list element.
+ */
+function getListItemBlocks(doc: HTMLDocument): ListItemBlock[] {
+    const listElements = doc.getElementsByClassName(LIST_CONTAINER_ELEMENT_CLASS_NAME);
+    const result: ListItemBlock[] = [];
+    let curListItemBlock: ListItemBlock;
+    for (let i = 0; i < listElements.length; i++) {
+        let curItem = listElements[i];
+        if (!curListItemBlock) {
+            curListItemBlock = createListItemBlock(curItem)
+        } else {
+            const { listItemContainers } = curListItemBlock;
+            const lastItemInCurBlock = listItemContainers[listItemContainers.length - 1];
+            if (curItem == lastItemInCurBlock.nextSibling
+                || getFirstLeafNode(curItem) == getNextLeafSibling(doc.body, lastItemInCurBlock)) {
+                listItemContainers.push(curItem);
+                curListItemBlock.endElement = curItem
+            } else {
+                curListItemBlock.endElement = lastItemInCurBlock;
+                result.push(curListItemBlock);
+                curListItemBlock = createListItemBlock(curItem);
+            }
+        }
+    }
+
+    if (curListItemBlock.listItemContainers.length > 0) {
+        result.push(curListItemBlock);
+    }
+
+    return result;
+}
+
+/**
+ * Flatten the list items, so that all the consecutive list items are under the same parent.
+ * @param doc Root element of that contains the element.
+ * @param listItemBlock The list item block needed to be flattened.
+ */
+function flattenListBlock(rootElement: Element, listItemBlock: ListItemBlock) {
+    const collapsedListItemSections = collapseNodes(rootElement, listItemBlock.startElement, listItemBlock.endElement, true);
+    collapsedListItemSections.forEach((section) => {
+        if (getTagOfNode(section.firstChild) == 'DIV') {
+            unwrap(section)
```

Comment:
are we ok to always unwrap? are we sure all DIVs here are newly added?

Reply:
Yes, each list item is wrap around by a Div. We need to remove the div for ease of processing before we start process the list item.